### DISCUSSION TITLE
Uploading a "large" PNG downscales it automatically? What are the limits?

### DISCUSSION THREAD

**Discussion by borekb** [excalidraw/excalidraw#4302] [Q&A]
I've tried to upload a PNG file to annotate it and it is a full-height screenshot of a website, a PNG size 2.1 MB and dimensions of **1500 × 8500 pixels**. This got uploaded as a highly **downscaled** image, roughly **100 x 550 pixels**.

It truly lost precision – making it larger shows the lack of pixels:

<img width="1568" alt="Screen Shot 2021-11-22 at 9 17 34" src="https://user-images.githubusercontent.com/101152/142826561-10abcd51-c426-411c-a196-5bbcdf1f7c59.png">

What kind of limit am I hitting? I found mentions of a limit in https://github.com/excalidraw/excalidraw/issues/19 but am not sure what exactly it is. Thanks for clarifying.

**Comment by borekb** [comment]
Hmm uploading to other web apps like Google Docs also downscales the image – am I hitting a limitation of some web API?

I can't share the original image but this screenshot of an Excalidraw issue behaves the same: [`github.com_excalidraw_excalidraw_issues_19.png`](https://user-images.githubusercontent.com/101152/142827660-bbc349d0-8b13-4cd8-9ff3-cc39a879ac24.png) (2.2 MB, ~1400 × 11000 pixels).

**Reply by borekb** [reply]
Diagrams-net give me this choice when uploading an image:

<img width="592" alt="Screen Shot 2021-11-22 at 9 34 08" src="https://user-images.githubusercontent.com/101152/142832443-bc5ed045-7a6f-4ce4-9042-0b7def9b1411.png">

When I choose "actual size", it fails for large images:

<img width="623" alt="Screen Shot 2021-11-22 at 9 29 47" src="https://user-images.githubusercontent.com/101152/142832498-8e15b1b1-b4ab-42a4-98e3-0b8459a977b7.png">

So probably some web API limitation.

**Comment by dwelle** [comment]
It's not necessarily a hard limit, but large images would have many problems, ranging from poor performance, to slow syncing, storage considerations etc. Performance is the biggest one here.

I'm planning to make optimizations to handle large PNGs, so I'll also look at cases like yours. If that image isn't private, can you upload it somewhere (maybe as a zip file in here, that shouldn't compress it), or send it to my DM on twitter. When I capture a screenshot of a large article (2000x8000px) I only get a 700kB PNG file, so I'd like to test similar dimensions like yours that are much larger file sizes.

**Reply by borekb** [reply]
I've posted an example of such image in https://github.com/excalidraw/excalidraw/discussions/4302#discussioncomment-1680008, can you reproduce it with it? Is it possible that it's just my computer (Chrome on macOS)?

I was briefly trying to google for this supposed limit as other apps also seem to limit my uploading capabilities (GDocs, diagrams.net) but I couldn't find anything conclusive.

So in Excalidraw's codebase, there aren't any explicit limits around this?

**Reply by dwelle** [reply]
> I've posted an example of such image in [#4302 (comment)](https://github.com/excalidraw/excalidraw/discussions/4302#discussioncomment-1680008), can you reproduce it with it? Is it possible that it's just my computer (Chrome on macOS)?

Oh, missed it. Good!

> So in Excalidraw's codebase, there aren't any explicit limits around this?

There are. We have a 2MB limit, after resizing the image to 1440px (largest axis).

The problem is there are wildly different use cases and we likely won't be able to come up with a "one size fits all" solution, no pun intended.

For example, if you want to import photos, they're a mix of large dimensions, large size (in this case you want to downsize), or small dimensions but large file size, such as 2000x1200px 5MB file (in which case you want to reduce file size).

In your case, you want to retain dimension more or less, but reduce file size. The problem is, our heuristic would pretty much never consider x8000px image to be a valid case to _not_ be resized. So I guess we'd have to give people the choice of what to do: resize vs compress. Compression will work mostly only for non-transparent PNGs though (so we can compress them to JPGs).

**Comment by mmdfl** [comment]
Same issue here. I solved my problem by slicing the image and jointing the pieces manually in excalidraw. I use postcron.com to do the slicing.

**Comment by CMCDragonkai** [comment]
This is really annoying, I wanted to use excalidraw to annotate a large image. Actually the image is just a screenshot of the website. And this makes it difficult to use for this purpose. I'm a paying plus member and I might as well use figma for this.

**Reply by CMCDragonkai** [reply]
And my image is only 1344x5445.

**Comment by austengary33** [comment]
Same issue. Any updates?
